File: MyPrintServiceNative/print_service_lib/src/main/java/com/example/print_service_lib/MyPrintService.java
package com.example.print_service_lib;

import android.printservice.PrintJob;
import android.printservice.PrintService;
import android.printservice.PrinterDiscoverySession;
import android.util.Log;

import androidx.annotation.Nullable;

public class MyPrintService extends PrintService {
    public static BackendPrinter[] sharedPrinters = new BackendPrinter[0];
    private static final String TAG = "MyPrintService";

    public static void printersToDiscover(BackendPrinter[] printers){
        sharedPrinters = printers;
    }

    @Nullable
    @Override
    protected PrinterDiscoverySession onCreatePrinterDiscoverySession() {
        Log.d(TAG, "Creating new printer discovery session");
        return new MyPrinterDiscoverySession(this, sharedPrinters);
    }

    @Override
    protected void onRequestCancelPrintJob(PrintJob printJob) {
        Log.d(TAG, "Cancelling print job: " + printJob.getId());
        printJob.cancel();
    }

    @Override
    protected void onPrintJobQueued(PrintJob printJob) {
        Log.d(TAG, "Print job queued: " + printJob.getId());
        printJob.start();
        printJob.complete();
    }
}

File: Old/MauiFrontend/MauiFrontend/Platforms/Android/MyPrintService.cs
using Android.App;
using Android.PrintServices;
using Android.Util;
using AndroidX.Annotations;


namespace MauiFrontend.Platforms.Android
{
    [Service(
        Exported = true,
        Permission = "android.permission.BIND_PRINT_SERVICE"
    )]
    [IntentFilter(new[] { "android.printservice.PrintService" })]
    public class MyPrintService : PrintService
    {
        const string TAG = "MyPrintService";

        // This static array is populated from your shared/.NET code.
        public static BackendPrinter[] SharedPrinters = new BackendPrinter[0];

        public MyPrintService()
        {
            // Required parameterless constructor for Android to instantiate the service.
        }

        // Called when Android wants you to create a new discovery session:
        protected override PrinterDiscoverySession OnCreatePrinterDiscoverySession()
        {
            Log.Debug(TAG, "Creating new printer discovery session (C#)");
            return new MyPrinterDiscoverySession(this, SharedPrinters);
        }

        protected override void OnRequestCancelPrintJob(PrintJob printJob)
        {
            Log.Debug(TAG, $"Cancelling print job: {printJob.Id?.ToString()}");
            printJob.Cancel();
        }

        protected override void OnPrintJobQueued(PrintJob printJob)
        {
            Log.Debug(TAG, $"Print job queued: {printJob.Id?.ToString()}");
            printJob.Start();
            printJob.Complete();
        }

        // C# helper to populate SharedPrinters from anywhere in your MAUI code:
        public void PrintersToDiscover(BackendPrinter[] printers)
        {
            SharedPrinters = printers;
            //OnCreatePrinterDiscoverySession();
        }
    }
}
